In [1]:
import pandas as pd

In [2]:
global_children = pd.read_csv('data/Child Mentality/global_student_mentalhealth_2003_2015.csv')

In [3]:
global_children.head()

Year LocationAbbr     LocationDesc DataSource              Topic  \
0  2010           NU             Niue       GSHS  Dietary Behaviors   
1  2003          UGB   Uganda (Urban)       GSHS  Dietary Behaviors   
2  2011           SI  Solomon Islands       GSHS  Dietary Behaviors   
3  2015           SH       Seychelles       GSHS      Mental Health   
4  2011           SI  Solomon Islands       GSHS  Dietary Behaviors   

        Subtopic            ShortQuestionText       Greater_Risk_Question  \
0    Body Weight                      Obesity                  Were obese   
1     Vegetables     No vegetable consumption      Did not eat vegetables   
2          Fruit  Fruit consumption >= 1 time                         NaN   
3  Mental Health                Close Friends  Did not have close friends   
4    Body Weight                   Overweight             Were overweight   

                                    Description Data_Value_Symbol  ... Age  \
0  >+2SD from the median for BMI by age and sex                 %  ...  13   
1          during the 30 days before the survey                 %  ...  13   
2          during the 30 days before the survey                 %  ...  13   
3                                           NaN                 %  ...  17   
4  >+1SD from the median for BMI by age and sex                 %  ...  13   

   GeoLocation  TopicID  SubtopicID  QuestionCode  LocationId StratID1  \
0          NaN      C05         C39      QNOBESEG         NaN       S1   
1          NaN      C05         C19            G8         NaN       S7   
2          NaN      C05         C41        QNFR1G         NaN       S8   
3          NaN      C36         C46           G27         NaN       S8   
4          NaN      C05         C39        QNOWTG         NaN       S8   

   StratID2  StratID3  StratificationType  
0        A2        NU     Western Pacific  
1        A2       UGB             African  
2        A2        SI     Western Pacific  
3        A6        SH             African  
4        A2        SI     Western Pacific  

[5 rows x 34 columns]

In [4]:
global_children['ShortQuestionText'].unique()

array(['Obesity', 'No vegetable consumption',
       'Fruit consumption >= 1 time', 'Close Friends', 'Overweight',
       'Underweight', 'No fruit consumption', 'Ever marjiuana use',
       'Fruit consumption >= 2 times', 'Ever sexual intercourse',
       'Condom use', 'Little vegetable consumption', 'Attempted suicide',
       'Considered suicide', 'Fruit consumption >= 3 times',
       'Little fruit consumption', 'Felt lonely',
       'Vegetable consumption >=1 time', 'PE attendance',
       'Fast food eating', 'Fast food eating >= 3 day',
       'Vegetable consumption >=2 times', 'Current cigarette use',
       'Parental tobacco use', 'Bullied', 'HIV/AIDS awareness',
       'Vegetable consumption >=3 times', 'Students kind and helpful',
       'Hunger', 'Fruits and Vegetables',
       'Talk with parents about HIV/AIDS', 'Seriously injured',
       'Bullied by hitting', 'Initiation of alcohol use',
       'Physically active >= 5 days', 'Washed hands before eating',
       'Multiple s

In [5]:
obese_children = global_children[(global_children['ShortQuestionText'] == 'Obesity') | 
                                (global_children['ShortQuestionText'] == 'Overweight')]

In [7]:
obese_children['Greater_Risk_Question'].unique()

array(['Were obese', 'Were overweight'], dtype=object)

In [8]:
obese_children.head()

Year LocationAbbr     LocationDesc DataSource              Topic  \
0   2010           NU             Niue       GSHS  Dietary Behaviors   
4   2011           SI  Solomon Islands       GSHS  Dietary Behaviors   
9   2007           MY          Myanmar       GSHS  Dietary Behaviors   
10  2010           MR          Morocco       GSHS  Dietary Behaviors   
18  2015           GT        Guatemala       GSHS  Dietary Behaviors   

       Subtopic ShortQuestionText Greater_Risk_Question  \
0   Body Weight           Obesity            Were obese   
4   Body Weight        Overweight       Were overweight   
9   Body Weight           Obesity            Were obese   
10  Body Weight           Obesity            Were obese   
18  Body Weight        Overweight       Were overweight   

                                     Description Data_Value_Symbol  ...  \
0   >+2SD from the median for BMI by age and sex                 %  ...   
4   >+1SD from the median for BMI by age and sex                 %  ...   
9   >+2SD from the median for BMI by age and sex                 %  ...   
10  >+2SD from the median for BMI by age and sex                 %  ...   
18  >+1SD from the median for BMI by age and sex                 %  ...   

      Age  GeoLocation  TopicID  SubtopicID  QuestionCode  LocationId  \
0      13          NaN      C05         C39      QNOBESEG         NaN   
4      13          NaN      C05         C39        QNOWTG         NaN   
9   13-15          NaN      C05         C39      QNOBESEG         NaN   
10     15          NaN      C05         C39      QNOBESEG         NaN   
18  13-15          NaN      C05         C39        QNOWTG         NaN   

   StratID1  StratID2  StratID3     StratificationType  
0        S1        A2        NU        Western Pacific  
4        S8        A2        SI        Western Pacific  
9        S1        A7        MY        South-East Asia  
10       S7        A4        MR  Eastern Mediterranean  
18       S7        A7        GT               Americas  

[5 rows x 34 columns]

In [9]:
obese_children.columns

Index(['Year', 'LocationAbbr', 'LocationDesc', 'DataSource', 'Topic',
       'Subtopic', 'ShortQuestionText', 'Greater_Risk_Question', 'Description',
       'Data_Value_Symbol', 'Data_Value_Type', 'Greater_Risk_Data_Value',
       'Greater_Risk_Data_Value_Footnote_Symbol',
       'Greater_Risk_Data_Value_Footnote', 'Greater_Risk_Low_Confidence_Limit',
       'Greater_Risk_High_Confidence_Limit', 'Lesser_Risk_Question',
       'Lesser_Risk_Data_Value', 'Lesser_Risk_Data_Value_Footnote_Symbol',
       'Lesser_Risk_Data_Value_Footnote', 'Lesser_Risk_High_Confidence_Limit',
       'Lesser_Risk_Low_Confidence_Limit', 'Sample_Size', 'Sex', 'Age',
       'GeoLocation', 'TopicID', 'SubtopicID', 'QuestionCode', 'LocationId',
       'StratID1', 'StratID2', 'StratID3', 'StratificationType'],
      dtype='object')

In [10]:
obese_children['Lesser_Risk_Question'].unique()

array([nan], dtype=object)

In [12]:
obese_children = obese_children[['Year', 'LocationDesc', 'Greater_Risk_Question', 'Data_Value_Symbol',
                'Greater_Risk_Data_Value', 'Sample_Size', 'Sex', 'Age']]

In [13]:
obese_children

Year                                LocationDesc  \
0       2010                                        Niue   
4       2011                             Solomon Islands   
9       2007                                     Myanmar   
10      2010                                     Morocco   
18      2015                                   Guatemala   
...      ...                                         ...   
278058  2005  United Arab Emirates (Abu Dhabi Emir Govt)   
278097  2012                             Argentina Chaco   
278124  2012                          Argentina San Juan   
278164  2010                                  UNRWA Gaza   
278182  2012                          Argentina Santa Fe   

       Greater_Risk_Question Data_Value_Symbol  Greater_Risk_Data_Value  \
0                 Were obese                 %                      NaN   
4            Were overweight                 %                      NaN   
9                 Were obese                 %                   0.6828   
10                Were obese                 %                   1.6316   
18           Were overweight                 %                  26.0481   
...                      ...               ...                      ...   
278058       Were overweight                 %                  40.7931   
278097            Were obese                 %                   6.7137   
278124            Were obese                 %                   7.3933   
278164            Were obese                 %                   4.5168   
278182            Were obese                 %                   6.6881   

        Sample_Size     Sex    Age  
0                12   Total     13  
4                80    Male     13  
9              1816   Total  13-15  
10              245  Female     15  
18             1641  Female  13-15  
...             ...     ...    ...  
278058          454   Total     13  
278097          109    Male     14  
278124          228   Total     14  
278164          864    Male  13-15  
278182          247    Male  13-15  

[6726 rows x 8 columns]

In [18]:
globe_internet = pd.read_csv('cleaned_data/global_internet_use_1960_2022_clean.csv')

In [19]:
globe_internet.head()

Country Name  1960  1961  1962  1963  1964  1965  1966  \
0                        Aruba   0.0   0.0   0.0   0.0   0.0   0.0   0.0   
1  Africa Eastern and Southern   0.0   0.0   0.0   0.0   0.0   0.0   0.0   
2                  Afghanistan   0.0   0.0   0.0   0.0   0.0   0.0   0.0   
3   Africa Western and Central   0.0   0.0   0.0   0.0   0.0   0.0   0.0   
4                       Angola   0.0   0.0   0.0   0.0   0.0   0.0   0.0   

   1967  1968  ...       2012       2013       2014       2015       2016  \
0   0.0   0.0  ...  74.000000  78.900000  83.780000  88.661227  93.542454   
1   0.0   0.0  ...   8.534750  10.126945  12.237716  14.593148  16.378025   
2   0.0   0.0  ...   5.454545   5.900000   7.000000   8.260000  11.000000   
3   0.0   0.0  ...  10.293655  12.577756  14.923441  18.179203  21.829266   
4   0.0   0.0  ...   7.700000  13.000000  21.400000  22.000000  23.200000   

        2017       2018       2019       2020       2021  
0  97.170000  97.170000  97.170000  97.170000  97.170000  
1  18.098060  21.653313  24.247500  26.200657  26.200657  
2  13.500000  16.800000  17.600000  18.400000  18.400000  
3  25.176342  29.305247  29.023474  34.093121  34.093121  
4  26.000000  29.000000  32.400000  36.000000  36.000000  

[5 rows x 63 columns]

In [20]:
globe_internet = globe_internet.melt(
    id_vars=['Country Name'], 
    var_name='Year', 
    value_name='Percent Internet Use'
)

In [22]:
globe_internet.head()

Country Name  Year  Percent Internet Use
0                        Aruba  1960                   0.0
1  Africa Eastern and Southern  1960                   0.0
2                  Afghanistan  1960                   0.0
3   Africa Western and Central  1960                   0.0
4                       Angola  1960                   0.0

In [25]:
globe_internet.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16492 entries, 0 to 16491
Data columns (total 3 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Country Name          16492 non-null  object 
 1   Year                  16492 non-null  object 
 2   Percent Internet Use  16492 non-null  float64
dtypes: float64(1), object(2)
memory usage: 386.7+ KB


In [27]:
obese_children['Year'] = obese_children['Year'].astype(str)

C:\Users\alecb\AppData\Local\Temp\ipykernel_3860\1081486423.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  obese_children['Year'] = obese_children['Year'].astype(str)


In [59]:
globe_internet_childobesity = pd.merge(obese_children, globe_internet, 
                               left_on = ('Year', 'LocationDesc'), right_on = ('Year', 'Country Name'), 
                               how = 'inner')

In [60]:
globe_internet_childobesity['LocationDesc'].unique()

array(['Solomon Islands', 'Myanmar', 'Morocco', 'Guatemala', 'Nauru',
       'Afghanistan', 'Indonesia', 'Uruguay', 'Guyana', 'Malaysia',
       'Oman', 'Jordan', 'Kuwait', 'India', 'Dominica', 'Samoa',
       'Bangladesh', 'Mongolia', 'El Salvador', 'Costa Rica', 'Iraq',
       'Argentina', 'Mauritania', 'Peru', 'Trinidad and Tobago', 'Uganda',
       'Pakistan', 'Jamaica', 'Belize', 'Namibia', 'Benin', 'Cambodia',
       'Sudan', 'Chile', 'Vietnam', 'Malawi', 'Barbados', 'Vanuatu',
       'Brunei Darussalam', 'Bolivia', 'Libya', 'Philippines', 'Lebanon',
       'Thailand', 'Seychelles', 'Djibouti', 'United Arab Emirates',
       'Maldives', 'Algeria', 'Kiribati', 'Tuvalu',
       'British Virgin Islands', 'Sri Lanka', 'Suriname', 'Tonga',
       'Ghana', 'Honduras'], dtype=object)

In [40]:
sorted(obese_children.LocationDesc.unique())

['Afghanistan',
 'Algeria',
 'Algeria Rural',
 'Algeria Urban',
 'Argentina',
 'Argentina Chaco',
 'Argentina Chubut',
 'Argentina Cordoba',
 'Argentina Corrientes',
 'Argentina Entre Rios',
 'Argentina Jujuy',
 'Argentina La Pampa',
 'Argentina Mendoza',
 'Argentina Neuquen',
 'Argentina Salta',
 'Argentina San Juan',
 'Argentina Santa Fe',
 'Argentina Santiago del Estero',
 'Bahamas',
 'Bangladesh',
 'Barbados',
 'Belize',
 'Benin',
 'Bolivia',
 'Bolivia Highland',
 'Bolivia Plains',
 'Bolivia Valley',
 'British Virgin Islands',
 'Brunei Darussalam',
 'Cambodia',
 'Cambodia Rural',
 'Cambodia Urban',
 'Chile',
 'Chile (Metropolitan Mun with Intervention)',
 'Chile (Metropolitan Mun without Intervention)',
 'Chile (Metropolitan Region)',
 'Chile (Region I)',
 'Chile (Region V)',
 'Chile (Region VIII Mun with Intervention)',
 'Chile (Region VIII Mun without Intervention)',
 'Chile (Region VIII)',
 'China (Beijing)',
 'China (Hangzhou)',
 'China (Wuhan)',
 'China (Wurumqi)',
 'Cook Isla

In [56]:
obese_children['LocationDesc'] = obese_children['LocationDesc'].str.split('(').str[0]
obese_children['LocationDesc'] = obese_children['LocationDesc'].str.split('-').str[0]

C:\Users\alecb\AppData\Local\Temp\ipykernel_3860\2970860120.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  obese_children['LocationDesc'] = obese_children['LocationDesc'].str.split('(').str[0]
C:\Users\alecb\AppData\Local\Temp\ipykernel_3860\2970860120.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  obese_children['LocationDesc'] = obese_children['LocationDesc'].str.split('-').str[0]


In [57]:
sorted(obese_children.LocationDesc.unique())

['Afghanistan',
 'Algeria',
 'Algeria Rural',
 'Algeria Urban',
 'Argentina',
 'Argentina Chaco',
 'Argentina Chubut',
 'Argentina Cordoba',
 'Argentina Corrientes',
 'Argentina Entre Rios',
 'Argentina Jujuy',
 'Argentina La Pampa',
 'Argentina Mendoza',
 'Argentina Neuquen',
 'Argentina Salta',
 'Argentina San Juan',
 'Argentina Santa Fe',
 'Argentina Santiago del Estero',
 'Bahamas',
 'Bangladesh',
 'Barbados',
 'Belize',
 'Benin',
 'Bolivia',
 'Bolivia Highland',
 'Bolivia Plains',
 'Bolivia Valley',
 'British Virgin Islands',
 'Brunei Darussalam',
 'Cambodia',
 'Cambodia Rural',
 'Cambodia Urban',
 'Chile',
 'Chile ',
 'China ',
 'Cook Islands',
 'Costa Rica',
 'Djibouti',
 'Dominica',
 'Ecuador ',
 'Egypt',
 'El Salvador',
 'Fiji Islands',
 'Ghana',
 'Ghana ',
 'Ghana Junior High',
 'Ghana Senior High',
 'Guatemala',
 'Guatemala ',
 'Guyana',
 'Honduras',
 'India',
 'Indonesia',
 'Indonesia ',
 'Iraq',
 'Jamaica',
 'Jordan',
 'Kiribati',
 'Kuwait',
 'Lebanon',
 'Lebanon ',
 'Liby

In [62]:
sorted(globe_internet['Country Name'].unique())

['Afghanistan',
 'Africa Eastern and Southern',
 'Africa Western and Central',
 'Albania',
 'Algeria',
 'American Samoa',
 'Andorra',
 'Angola',
 'Antigua and Barbuda',
 'Arab World',
 'Argentina',
 'Armenia',
 'Aruba',
 'Australia',
 'Austria',
 'Azerbaijan',
 'Bahamas, The',
 'Bahrain',
 'Bangladesh',
 'Barbados',
 'Belarus',
 'Belgium',
 'Belize',
 'Benin',
 'Bermuda',
 'Bhutan',
 'Bolivia',
 'Bosnia and Herzegovina',
 'Botswana',
 'Brazil',
 'British Virgin Islands',
 'Brunei Darussalam',
 'Bulgaria',
 'Burkina Faso',
 'Burundi',
 'Cabo Verde',
 'Cambodia',
 'Cameroon',
 'Canada',
 'Caribbean small states',
 'Cayman Islands',
 'Central African Republic',
 'Central Europe and the Baltics',
 'Chad',
 'Channel Islands',
 'Chile',
 'China',
 'Colombia',
 'Comoros',
 'Congo, Dem. Rep.',
 'Congo, Rep.',
 'Costa Rica',
 "Cote d'Ivoire",
 'Croatia',
 'Cuba',
 'Curacao',
 'Cyprus',
 'Czechia',
 'Denmark',
 'Djibouti',
 'Dominica',
 'Dominican Republic',
 'Early-demographic dividend',
 'East

In [61]:
globe_internet_childobesity.corr()

Greater_Risk_Data_Value  Sample_Size  \
Greater_Risk_Data_Value                 1.000000    -0.039667   
Sample_Size                            -0.039667     1.000000   
Percent Internet Use                    0.371956     0.085989   

                         Percent Internet Use  
Greater_Risk_Data_Value              0.371956  
Sample_Size                          0.085989  
Percent Internet Use                 1.000000

#### The correlation coefficient (on the global scale) for childhood obesity and internet users is 0.372

In [58]:
obese_children.to_csv('global_obese_children.csv', index = False)

In [37]:
globe_internet.to_csv('global_internet_use.csv', index = False)